resample/bootstrap.py

```diff
@@ -356,3 +366,20 @@ def _confidence_interval_bca(
 
     quant = quantile_function_gen(thetas)
     return quant(p_low), quant(p_high)
+
+
+def _jackknife_after_bootstrap(
+    fn: Callable, thetas: np.ndarray, resampled: np.ndarray, sample: np.ndarray
+) -> Tuple[np.ndarray, np.ndarray]:
+    # computes the jackknife-after-bootstrap, as described in the book Efron &
+    # Tibshirani, An introduction to the bootstrap.
+    n = len(sample)
+    jack: List[List[int]] = [[] for i in range(n)]
+    for i, b in enumerate(resampled):
+        for j, s in enumerate(sample):
+            if np.any(s == b):
```

Comment:
I think this may need to be done at a lower level if we want to do it correctly. We can't for instance rely on equality because that breaks as soon as we have ties in the data.

Reply:
Hm, I think you are right. I thought that ties do not matter, but now that I think about it more, I think you are correct.

Reply:
Maybe the resampling generators could optionally yield a tuple of arrays, one for the underlying data and another for the indexes, with that behavior being controlled by a boolean keyword `return_indexes` / `return_indices` or something similar. Then the jackknife after bootstrap could access those indexes to figure out which replicates contain which data points.